src/mon/OSDMonitor.cc

```diff
@@ -5819,12 +5855,15 @@ bool OSDMonitor::prepare_command_impl(MMonCommand *m,
 	}
 	goto reply;
       }
-      i = pending_inc.identify_osd(uuid);
-      if (i >= 0) {
+      if (pending_inc.new_state.count(i)) {
 	// osd is about to exist
 	wait_for_finished_proposal(new C_RetryMessage(this, m));
 	return true;
       }
+      // raise max_osd
+      if (osdmap.get_max_osd() <= i && pending_inc.new_max_osd <= i)
```

Comment:
For me, the condition 'osdmap.get_max_osd() > i && pending_inc.new_max_osd <= i' means that we already have a pending update that increases max_osd (e.g. from 'ceph osd create uuid lage_id') and are processing another update that allocates an ID in the gap. So in this case we should not increase max_osd.